packages/block-library/src/post-terms/index.php

```diff
@@ -49,13 +49,19 @@ function render_block_core_post_terms( $attributes, $content, $block ) {
 		$suffix = '<span class="wp-block-post-terms__suffix">' . $attributes['suffix'] . '</span>' . $suffix;
 	}
 
-	return get_the_term_list(
+	$result = get_the_term_list(
 		$block->context['postId'],
 		$attributes['term'],
 		wp_kses_post( $prefix ),
 		'<span class="wp-block-post-terms__separator">' . esc_html( $separator ) . '</span>',
 		wp_kses_post( $suffix )
 	);
+
+	if ( ! empty( $attributes['noLink'] ) ) {
+		return preg_replace( '#<a.*?>(.*?)</a>#i', '<span class="wp-block-post-terms__name">\1</span>', $result );
```

Comment:
**Question**, would it be better to use get_the_terms() here instead of get_the_term_list with a preg_replace to remove the links?
Both can be filtered, right?. I don't think we can trust the markup to be the default?